src/main/ui/app/route/SearchPage.jsx

```diff
@@ -30,28 +35,56 @@ export default class SearchPage extends Component {
         }
     }
 
-    doSearch() {
-        const query = this.props.location.query.q;
-        const type = this.props.location.query.type;
-	const tag = this.props.location.query.tag;
-	const searchKey = `type:${type ? type : "all"} query:${query ? query : ""} tag:${tag ? tag : ""}`;
+    normalizeQuery(query) {
+        //Basic Regex Matching to fix query inconsistencies
+        //Removing all extra spaces i.e. two or more
+        query = query.replace(/\s+/g,' ').trim();
+        //Removing all spaces to the left of search keys
+        query = query.replace(new RegExp("\\s+:","g"),":");
+        //Removing all spaces to the right of search keys
+        query = query.replace(new RegExp(":\\s+","g"),":");
+        return query;
+    }
 
+    doSearch() {
+        const searchKey = this.props.location.query.q;
+        let query = searchKey.toLowerCase();
         if (this.state.latestSearch === searchKey) {
             return;
         }
 
         this.handleStartSearch(searchKey);
 
         let url = "/api/v1/addon?";
-        if (type) {
-            url += "type=" + type;
+
+        query = this.normalizeQuery(query);
+
+        // Check if query is an advanced query. We want to use the Parser only if the query is advanced
+        if (query.includes(":") || query.includes("-")) {
+            let searchQueryObj = searchQuery.parse(query, options);
+            Object.keys(searchQueryObj).forEach(function (key) {
+                if (searchQueryObj[key]) {
+                    if (key === "text" || key === "query") {
+                        key === "text" ? url += "&q=" + searchQueryObj[key].join(" ") : url += "&q=" + searchQueryObj[key];
+                    }
+                    else if (key === "type" || key === "status") {
+                        url += "&" + key + "=" + searchQueryObj[key].toUpperCase();
+                    }
+                    else if (key === "exclude") {
+                        if (searchQueryObj[key].text) {
+                            url += "&exclude=" + searchQueryObj[key].text;
+                        }
+                    }
+                    else {
+                        url += "&" + key + "=" + searchQueryObj[key];
+                    }
+                }
+            });
+        }
+        else {
+            url += "&q=" + query;
         }
-        if (query) {
-			url += "&q=" + query;
-		}
-		if (tag) {
-			url += "&tag=" + tag;
```

Comment:
This change implies there should be a change to the URL of the link for searching by tag, right? I don't see it changed in this PR.